Milestone 1: Awesome books: with ES6

carloshs1994 opened the pull request:
# In this milestone:
✅  Set up the repo using npm
✅  Recreate app using ES6
✅  Display the current date below the navigation bar. - For this requirement, we used a library - Luxon
<img src="https://media1.giphy.com/media/mp1JYId8n0t3y/giphy.gif"/>

carloshs1994 committed:
Set up files with previous project

joseheco committed:
export function localstorage

Aduda-Boaz committed:
Append books module

carloshs1994 committed:
Add singleBook.js module

joseheco committed:
export luxon js file

Aduda-Boaz committed:
Add date and time luxon

carloshs1994 committed:
Fix date-and-time bug

carloshs1994 committed:
Add linters to repo

carloshs1994 committed:
Fix linters bug

joseheco committed:
fix code eslinter done

joseheco committed:
fix code eslinter done

joseheco committed:
fix code eslinter js

joseheco committed:
fix code eslinter js

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+//     this.title = title;
+//     this.author = author;
+//   }
+// }
```
- [x] **Required** It is unprofessional to push commented code. This is against the best practices at Microverse and further in your career. Please remove all commented code from the codebase.

brytebee commented on modules/dist/addbooks.dev.js:
```diff
@@ -0,0 +1,24 @@
+});
+exports.appendBooksToList = appendBooksToList;
+
+var _index = require("../modules/index.js");
```
- [x] **Required** To comply with ES6 which is the focus of this and upcoming projects, please replace all variables declared with the `var` keyword with either `const` or `let` keywords. See this [resource](https://evertpot.com/javascript-let-const/) to understand more. Hope it makes sense 😄

joseheco replied:
this file is the dist file, we dont create that file is auto create , you can see its a addbooks.dev.js, the real file that we work is on ./module/addbooks.

brytebee commented on modules/dist/addbooks.dev.js:
```diff
@@ -0,0 +1,24 @@
+  _index.list.innerHTML = '';
+
+  _index.books.bookList.forEach(function (book, index) {
+    var li = document.createElement('li');
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+
+var _addbooks = require("./addbooks.js");
+
+var _singleBook = require("./singleBook.js");
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+var listSection = document.getElementById('list-section');
+var formSection = document.getElementById('form-section');
+var contactSection = document.getElementById('contact-section');
+var DateTime = luxon.DateTime;
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+  return Books;
+}();
+
+var books = new Books();
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+  for (var i = 0; i < removeButtons.length; i += 1) {
+    var button = removeButtons[i];
+    button.addEventListener('click', function (event) {
+      var element = event.target.parentElement.firstElementChild.firstElementChild.innerText;
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+var removeBook = function removeBook() {
+  var removeButtons = document.getElementsByClassName('remove');
+
+  for (var i = 0; i < removeButtons.length; i += 1) {
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+exports.updateDomAndLocalStorage = updateDomAndLocalStorage;
+
+var removeBook = function removeBook() {
+  var removeButtons = document.getElementsByClassName('remove');
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on modules/dist/index.dev.js:
```diff
@@ -0,0 +1,147 @@
+
+form.addEventListener('submit', function (event) {
+  event.preventDefault();
+  var newBook = new _singleBook.Book(title.value, author.value);
```
- [x] **Required:** Same as above. Replace `var` with `const` (if the variable is never mutated) or `let` (if the variable is mutated).

brytebee commented on .gitignore:
```diff
@@ -0,0 +1 @@
+node_modules
```
- [x] **Required:** Let's ensure the node_modules are properly ignored because some of them were still pushed.

carloshs1994 replied:
Done! we pushed by accident before we created the .gitignore file

brytebee reviewed (changes requested):
Hi, @Team 👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- ✔️ Amazing UI and design.
- ✔️ Proper git messages and GitHub flow
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/VL48WGMDjD64umCEkv/giphy.gif)

# Required Changes ♻️

- [ ] **Required** Properly document your work by completing the ReadMe file. Add the project snapshots and live link (optional). Ensure to complete the USAGE, PREREQUISITE and SETUP sections as well.
- [ ] **Required:** For some reason, it is only the Home page (All Awesome books) that loads on my PC. The form and contact pages do not load on click. Let's fix this.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Aduda-Boaz commented:
Hello @brytebee, just a quick question, I see you pointed at the files in the dist folder, which were created when we saved our codes in the module folder we created. Do we really need to change it since in my opinion they were created automatically?

carloshs1994 commented:
I saw that as well @Aduda-Boaz, I'll try to remove them from the repo because those are not relevant

elisha2kyakpo1 commented:
Hello @Aduda-Boaz I see that you made another submission like you stated, let's make the changes suggested by @brytebee.
As for the auto generated code in dist folder, you may leave a note for the next reviewer, and by the way I would like to know how it was generated since the project is not using any compiler like webpack?

carloshs1994 committed:
Delete node_modules file

It was pushed by mistake before we set up linters

carloshs1994 committed:
Fix suggested changes:
- Update the README.md file
- Add link to live demo
- Remove the node_modules from the repo and we are ignoring it using .gitignore
- remove .dev.js files generated by npm by mistake

After all this the 'Add Book' and 'contact' sections work fine

carloshs1994 commented:
> Hello @Aduda-Boaz I see that you made another submission like you stated, let's make the changes suggested by @brytebee. As for the auto generated code in dist folder, you may leave a note for the next reviewer, and by the way I would like to know how it was generated since the project is not using any compiler like webpack?

We are not sure but we delete the folder so It does not create further issues 😃

Mihndim2020 commented on modules/addbooks.js:
```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line import/no-cycle
```
- [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on modules/index.js:
```diff
@@ -0,0 +1,114 @@
+//  eslint-disable-next-line import/no-cycle
```
- [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on modules/index.js:
```diff
@@ -0,0 +1,114 @@
+//  eslint-disable-next-line import/no-cycle
+import { addToLocalStorage, getFromLocalStorage } from './localstorage.js';
+//  eslint-disable-next-line import/no-cycle
```
- [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on modules/localstorage.js:
```diff
@@ -0,0 +1,17 @@
+//  eslint-disable-next-line import/no-cycle
```
- [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on modules/index.js:
```diff
@@ -0,0 +1,114 @@
+import appendBooksToList from './addbooks.js';
+import Book from './singleBook.js';
+
+const { DateTime } = luxon; //  eslint-disable-line
```
- [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

carloshs1994 replied:
> * [OPTIONAL]  `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice. Instead of disabling `eslint`, it will be nice to fix the issue.

Thanks for the suggestions @Mihndim2020, we'll fix it!

Mihndim2020 reviewed (approved):
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi TEAM, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

![Congratulations-Memes-FEATURED](https://user-images.githubusercontent.com/65852621/133696300-b506e476-dc1a-443e-935a-c6f434e788b6.jpg)

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

rloterh reviewed (commented):
Hi Boaz,  

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: . Congratulations!  🎉 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
